trivial code does not compile on windows (C names conflict with V names)

**V version:** 
```
V 0.1.25 5142747
gcc version 8.1.0 (x86_64-posix-seh-rev0, Built by MinGW-W64 project)
```
**OS:** Win10

**What did you do?**
compile this trivial source file:
```
fn main() {
	log('nox')
}

fn log(line string) {
	println(line)
}
```


**What did you expect to see?**
no error.
Note: older version of v compiles without errors
`V 0.1.25 d499116` (approx 2020-02-15)
 
**What did you see instead?**
```
c:\dev_alt\vlang\test>v -cg trivial_error.v
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:159: warning: "UNICODE" redefined
 #define UNICODE

<command-line>: note: this is the location of the previous definition
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c: In function 'print':
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:1631:58: warning: passing argument 4 of 'WriteConsoleW' from incompatible pointer type [-Wincompatible-pointer-types]
  WriteConsole ( output_handle ,  wide_str ,  wide_len ,  & bytes_written ,  0 ) ;
                                                          ^~~~~~~~~~~~~~~
In file included from c:/dev_alt/mingw64/x86_64-w64-mingw32/include/windows.h:74,
                 from C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:160:
c:/dev_alt/mingw64/x86_64-w64-mingw32/include/wincon.h:260:122: note: expected 'LPDWORD' {aka 'long unsigned int *'} but argument is of type 'int *'
   WINBASEAPI WINBOOL WINAPI WriteConsoleW(HANDLE hConsoleOutput,CONST VOID *lpBuffer,DWORD nNumberOfCharsToWrite,LPDWORD lpNumberOfCharsWritten,LPVOID lpReserved);
                                                                                                                  ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:1634:58: warning: passing argument 4 of 'WriteFile' from incompatible pointer type [-Wincompatible-pointer-types]
  WriteFile ( output_handle , (char*) s .str ,  s .len ,  & bytes_written ,  0 ) ;
                                                          ^~~~~~~~~~~~~~~
In file included from c:/dev_alt/mingw64/x86_64-w64-mingw32/include/winbase.h:18,
                 from c:/dev_alt/mingw64/x86_64-w64-mingw32/include/windows.h:70,
                 from C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:160:
c:/dev_alt/mingw64/x86_64-w64-mingw32/include/fileapi.h:175:109: note: expected 'LPDWORD' {aka 'long unsigned int *'} but argument is of type 'int *'
   WINBASEAPI WINBOOL WINAPI WriteFile (HANDLE hFile, LPCVOID lpBuffer, DWORD nNumberOfBytesToWrite, LPDWORD lpNumberOfBytesWritten, LPOVERLAPPED lpOverlapped);
                                                                                                     ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c: In function 'strconv__atof64':
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:5031:65: warning: initialized field overwritten [-Woverride-init]
 strconv__Float64u res= (strconv__Float64u) { .f =  0.0 , .u =   ((u64)( 0 ) ) } ;
                                                                 ^
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:5031:65: note: (near initialization for '(anonymous).u')
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c: In function 'strconv_dot_ftoa__f32_to_str':
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:5999:84: warning: initialized field overwritten [-Woverride-init]
 strconv_dot_ftoa__Uf32 u1= (strconv_dot_ftoa__Uf32) { .f =   ((f32)( 0 ) ) , .u =  0 } ;
                                                                                    ^
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:5999:84: note: (near initialization for '(anonymous).u')
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c: In function 'strconv_dot_ftoa__f64_to_str':
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:6262:84: warning: initialized field overwritten [-Woverride-init]
 strconv_dot_ftoa__Uf64 u1= (strconv_dot_ftoa__Uf64) { .f =   ((f64)( 0 ) ) , .u =  0 } ;
                                                                                    ^
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:6262:84: note: (near initialization for '(anonymous).u')
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c: In function 'main__main':
C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:6612:8: error: incompatible type for argument 1 of 'log'
  log ( "nox" ) ;
        ^~~~~
In file included from C:\Users\alisurbo\AppData\Local\Temp\v\trivial_error.tmp.c:369:
c:/dev_alt/mingw64/x86_64-w64-mingw32/include/math.h:194:29: note: expected 'double' but argument is of type 'char *'
   double __cdecl log(double _X);
                      ~~~~~~~^~
V error:
==================
C error. This should never happen.

V compiler version: V 0.1.25 5142747
Host OS: Windows
Target OS: Windows

If you were not working with C interop and are not sure about what's happening,
please put the whole output in a pastebin and contact us through the following ways with a link to the pastebin:
- Raise an issue on GitHub: https://github.com/vlang/v/issues/new/choose
- Ask a question in #help on Discord: https://discord.gg/vlang
```

--- a/examples/sokol/drawing.v
+++ b/examples/sokol/drawing.v
@@ -28,13 +28,13 @@ fn main() {
 
 fn init(user_data voidptr) {
 	desc := sg_desc{
-		mtl_device: C.sapp_metal_get_device()
-		mtl_renderpass_descriptor_cb: C.sapp_metal_get_renderpass_descriptor
-		mtl_drawable_cb: C.sapp_metal_get_drawable
-		d3d11_device: C.sapp_d3d11_get_device()
-		d3d11_device_context: C.sapp_d3d11_get_device_context()
-		d3d11_render_target_view_cb: C.sapp_d3d11_get_render_target_view
-		d3d11_depth_stencil_view_cb: C.sapp_d3d11_get_depth_stencil_view
+		mtl_device: sapp.metal_get_device()
+		mtl_renderpass_descriptor_cb: sapp.metal_get_renderpass_descriptor
+		mtl_drawable_cb: sapp.metal_get_drawable
+		d3d11_device: sapp.d3d11_get_device()
+		d3d11_device_context: sapp.d3d11_get_device_context()
+		d3d11_render_target_view_cb: sapp.d3d11_get_render_target_view
+		d3d11_depth_stencil_view_cb: sapp.d3d11_get_depth_stencil_view
 	}
 	gfx.setup(&desc)
 	sgl_desc := sgl_desc_t{}
@@ -75,9 +75,9 @@ fn draw_hollow_rect(x, y, w, h f32) {
 
 fn draw_filled_rect(x, y, w, h f32) {
 	sgl.begin_quads()
-	C.sgl_v2f(x, y)
-	C.sgl_v2f(x + w, y)
-	C.sgl_v2f(x + w, y + h)
-	C.sgl_v2f(x, y + h)
+	sgl.v2f(x, y)
+	sgl.v2f(x + w, y)
+	sgl.v2f(x + w, y + h)
+	sgl.v2f(x, y + h)
 	sgl.end()
 }

--- a/examples/sokol/drawing.v
+++ b/examples/sokol/drawing.v
@@ -29,12 +29,12 @@ fn main() {
 fn init(user_data voidptr) {
 	desc := sg_desc{
 		mtl_device: C.sapp_metal_get_device()
-		mtl_renderpass_descriptor_cb: sapp_metal_get_renderpass_descriptor
-		mtl_drawable_cb: sapp_metal_get_drawable
-		d3d11_device: sapp_d3d11_get_device()
-		d3d11_device_context: sapp_d3d11_get_device_context()
-		d3d11_render_target_view_cb: sapp_d3d11_get_render_target_view
-		d3d11_depth_stencil_view_cb: sapp_d3d11_get_depth_stencil_view
+		mtl_renderpass_descriptor_cb: C.sapp_metal_get_renderpass_descriptor
+		mtl_drawable_cb: C.sapp_metal_get_drawable
+		d3d11_device: C.sapp_d3d11_get_device()
+		d3d11_device_context: C.sapp_d3d11_get_device_context()
+		d3d11_render_target_view_cb: C.sapp_d3d11_get_render_target_view
+		d3d11_depth_stencil_view_cb: C.sapp_d3d11_get_depth_stencil_view
 	}
 	gfx.setup(&desc)
 	sgl_desc := sgl_desc_t{}
@@ -75,9 +75,9 @@ fn draw_hollow_rect(x, y, w, h f32) {
 
 fn draw_filled_rect(x, y, w, h f32) {
 	sgl.begin_quads()
-	sgl_v2f(x, y)
-	sgl_v2f(x + w, y)
-	sgl_v2f(x + w, y + h)
-	sgl_v2f(x, y + h)
+	C.sgl_v2f(x, y)
+	C.sgl_v2f(x + w, y)
+	C.sgl_v2f(x + w, y + h)
+	C.sgl_v2f(x, y + h)
 	sgl.end()
 }

--- a/examples/sokol/fonts.v
+++ b/examples/sokol/fonts.v
@@ -42,13 +42,13 @@ fn init(user_data voidptr) {
 	mut state := &AppState(user_data)
 	// dont actually alocate this on the heap in real life
 	gfx.setup(&sg_desc{
-		mtl_device: C.sapp_metal_get_device()
-		mtl_renderpass_descriptor_cb: sapp_metal_get_renderpass_descriptor
-		mtl_drawable_cb: sapp_metal_get_drawable
-		d3d11_device: sapp_d3d11_get_device()
-		d3d11_device_context: sapp_d3d11_get_device_context()
-		d3d11_render_target_view_cb: sapp_d3d11_get_render_target_view
-		d3d11_depth_stencil_view_cb: sapp_d3d11_get_depth_stencil_view
+		mtl_device: sapp.metal_get_device()
+		mtl_renderpass_descriptor_cb: sapp.metal_get_renderpass_descriptor
+		mtl_drawable_cb: sapp.metal_get_drawable
+		d3d11_device: sapp.d3d11_get_device()
+		d3d11_device_context: sapp.d3d11_get_device_context()
+		d3d11_render_target_view_cb: sapp.d3d11_get_render_target_view
+		d3d11_depth_stencil_view_cb: sapp.d3d11_get_depth_stencil_view
 	})
 	s := &C.sgl_desc_t{}
 	C.sgl_setup(s)
@@ -64,7 +64,7 @@ fn frame(user_data voidptr) {
 	t := time.ticks()
 	mut state := &AppState(user_data)
 	state.render_font()
-	gfx.begin_default_pass(&state.pass_action, sapp_width(), sapp_height())
+	gfx.begin_default_pass(&state.pass_action, sapp.width(), sapp.height())
 	sgl.draw()
 	gfx.end_pass()
 	gfx.commit()
@@ -98,43 +98,43 @@ fn (state &AppState) render_font() {
 	dy += lh
 	C.fonsSetColor(state.fons, white)
 	dx = C.fonsDrawText(state.fons, dx, dy, c'The quick ', C.NULL)
-	fonsSetFont(state.fons, state.font_normal)
-	fonsSetSize(state.fons, 48.0)
-	fonsSetColor(state.fons, brown)
-	dx = fonsDrawText(state.fons, dx, dy, c'brown ', C.NULL)
-	fonsSetFont(state.fons, state.font_normal)
-	fonsSetSize(state.fons, 24.0)
-	fonsSetColor(state.fons, white)
-	dx = fonsDrawText(state.fons, dx, dy, c'fox ', C.NULL)
+	C.fonsSetFont(state.fons, state.font_normal)
+	C.fonsSetSize(state.fons, 48.0)
+	C.fonsSetColor(state.fons, brown)
+	dx = C.fonsDrawText(state.fons, dx, dy, c'brown ', C.NULL)
+	C.fonsSetFont(state.fons, state.font_normal)
+	C.fonsSetSize(state.fons, 24.0)
+	C.fonsSetColor(state.fons, white)
+	dx = C.fonsDrawText(state.fons, dx, dy, c'fox ', C.NULL)
 	dx = sx
 	dy += lh * 1.2
-	fonsSetSize(state.fons, 20.0)
-	fonsSetFont(state.fons, state.font_normal)
-	fonsSetColor(state.fons, blue)
-	fonsDrawText(state.fons, dx, dy, c'Now is the time for all good men to come to the aid of the party.', C.NULL)
+	C.fonsSetSize(state.fons, 20.0)
+	C.fonsSetFont(state.fons, state.font_normal)
+	C.fonsSetColor(state.fons, blue)
+	C.fonsDrawText(state.fons, dx, dy, c'Now is the time for all good men to come to the aid of the party.', C.NULL)
 	dx = 300
 	dy = 350
-	fonsSetAlign(state.fons, C.FONS_ALIGN_LEFT | C.FONS_ALIGN_BASELINE)
-	fonsSetSize(state.fons, 60.0)
-	fonsSetFont(state.fons, state.font_normal)
-	fonsSetColor(state.fons, white)
-	fonsSetSpacing(state.fons, 5.0)
-	fonsSetBlur(state.fons, 6.0)
-	fonsDrawText(state.fons, dx, dy, c'Blurry...', C.NULL)
+	C.fonsSetAlign(state.fons, C.FONS_ALIGN_LEFT | C.FONS_ALIGN_BASELINE)
+	C.fonsSetSize(state.fons, 60.0)
+	C.fonsSetFont(state.fons, state.font_normal)
+	C.fonsSetColor(state.fons, white)
+	C.fonsSetSpacing(state.fons, 5.0)
+	C.fonsSetBlur(state.fons, 6.0)
+	C.fonsDrawText(state.fons, dx, dy, c'Blurry...', C.NULL)
 	dx = 300
 	dy += 50.0
-	fonsSetSize(state.fons, 28.0)
-	fonsSetFont(state.fons, state.font_normal)
-	fonsSetColor(state.fons, white)
-	fonsSetSpacing(state.fons, 0.0)
-	fonsSetBlur(state.fons, 3.0)
-	fonsDrawText(state.fons, dx, dy + 2, c'DROP SHADOW', C.NULL)
-	fonsSetColor(state.fons, black)
-	fonsSetBlur(state.fons, 0)
-	fonsDrawText(state.fons, dx, dy, c'DROP SHADOW', C.NULL)
-	fonsSetSize(state.fons, 18.0)
-	fonsSetFont(state.fons, state.font_normal)
-	fonsSetColor(state.fons, white)
+	C.fonsSetSize(state.fons, 28.0)
+	C.fonsSetFont(state.fons, state.font_normal)
+	C.fonsSetColor(state.fons, white)
+	C.fonsSetSpacing(state.fons, 0.0)
+	C.fonsSetBlur(state.fons, 3.0)
+	C.fonsDrawText(state.fons, dx, dy + 2, c'DROP SHADOW', C.NULL)
+	C.fonsSetColor(state.fons, black)
+	C.fonsSetBlur(state.fons, 0)
+	C.fonsDrawText(state.fons, dx, dy, c'DROP SHADOW', C.NULL)
+	C.fonsSetSize(state.fons, 18.0)
+	C.fonsSetFont(state.fons, state.font_normal)
+	C.fonsSetColor(state.fons, white)
 	dx = 50
 	dy = 350
 	line(dx - 10, dy, dx + 250, dy)

--- a/vlib/compiler/expression.v
+++ b/vlib/compiler/expression.v
@@ -366,7 +366,7 @@ fn (p mut Parser) name_expr() string {
 	}
 	// Unknown name, try prepending the module name to it
 	// TODO perf
-	else if !p.table.known_type(name) && !p.table.known_fn(name) && !p.table.known_const(name) && !is_c {
+	else if !p.table.known_type(name) && !p.known_fn_in_mod(name) && !p.table.known_const(name) && !is_c {
 		name = p.prepend_mod(name)
 	}
 	// re-check

--- a/vlib/compiler/table.v
+++ b/vlib/compiler/table.v
@@ -384,6 +384,16 @@ fn (t &Table) known_fn(name string) bool {
 	return true
 }
 
+fn (p &Parser) known_fn_in_mod(name string) bool {
+	existing_fn := p.table.find_fn(name) or {
+		return false
+	}
+	if (existing_fn.mod == p.mod || existing_fn.mod == 'builtin') {
+		return true
+	}
+	return false
+}
+
 fn (t &Table) known_const(name string) bool {
 	_ = t.find_const(name) or {
 		return false

--- a/vlib/gg2/gg.v
+++ b/vlib/gg2/gg.v
@@ -57,12 +57,12 @@ __global g_font_path string
 fn init_sokol_window() {
 	desc := sg_desc{
 		mtl_device: C.sapp_metal_get_device()
-		mtl_renderpass_descriptor_cb: sapp_metal_get_renderpass_descriptor
-		mtl_drawable_cb: sapp_metal_get_drawable
-		d3d11_device: sapp_d3d11_get_device()
-		d3d11_device_context: sapp_d3d11_get_device_context()
-		d3d11_render_target_view_cb: sapp_d3d11_get_render_target_view
-		d3d11_depth_stencil_view_cb: sapp_d3d11_get_depth_stencil_view
+		mtl_renderpass_descriptor_cb: C.sapp_metal_get_renderpass_descriptor
+		mtl_drawable_cb: C.sapp_metal_get_drawable
+		d3d11_device: C.sapp_d3d11_get_device()
+		d3d11_device_context: C.sapp_d3d11_get_device_context()
+		d3d11_render_target_view_cb: C.sapp_d3d11_get_render_target_view
+		d3d11_depth_stencil_view_cb: C.sapp_d3d11_get_depth_stencil_view
 	}
 	gfx.setup(&desc)
 	sgl_desc := sgl_desc_t{}
@@ -138,20 +138,20 @@ pub fn (gg &GG) run() {
 pub fn (ctx &GG) draw_rect(x, y, w, h f32, c gx.Color) {
 	sgl.c4b(c.r, c.g, c.b, 128)
 	sgl.begin_quads()
-	sgl_v2f(x, y)
-	sgl_v2f(x + w, y)
-	sgl_v2f(x + w, y + h)
-	sgl_v2f(x, y + h)
+	C.sgl_v2f(x, y)
+	C.sgl_v2f(x + w, y)
+	C.sgl_v2f(x + w, y + h)
+	C.sgl_v2f(x, y + h)
 	sgl.end()
 }
 
 pub fn draw_rect(x, y, w, h f32, c gx.Color) {
 	sgl.c4b(c.r, c.g, c.b, 128)
 	sgl.begin_quads()
-	sgl_v2f(x, y)
-	sgl_v2f(x + w, y)
-	sgl_v2f(x + w, y + h)
-	sgl_v2f(x, y + h)
+	C.sgl_v2f(x, y)
+	C.sgl_v2f(x + w, y)
+	C.sgl_v2f(x + w, y + h)
+	C.sgl_v2f(x, y + h)
 	sgl.end()
 }
 

--- a/vlib/gg2/gg.v
+++ b/vlib/gg2/gg.v
@@ -56,13 +56,13 @@ __global g_font_path string
 
 fn init_sokol_window() {
 	desc := sg_desc{
-		mtl_device: C.sapp_metal_get_device()
-		mtl_renderpass_descriptor_cb: C.sapp_metal_get_renderpass_descriptor
-		mtl_drawable_cb: C.sapp_metal_get_drawable
-		d3d11_device: C.sapp_d3d11_get_device()
-		d3d11_device_context: C.sapp_d3d11_get_device_context()
-		d3d11_render_target_view_cb: C.sapp_d3d11_get_render_target_view
-		d3d11_depth_stencil_view_cb: C.sapp_d3d11_get_depth_stencil_view
+		mtl_device: sapp.metal_get_device()
+		mtl_renderpass_descriptor_cb: sapp.metal_get_renderpass_descriptor
+		mtl_drawable_cb: sapp.metal_get_drawable
+		d3d11_device: sapp.d3d11_get_device()
+		d3d11_device_context: sapp.d3d11_get_device_context()
+		d3d11_render_target_view_cb: sapp.d3d11_get_render_target_view
+		d3d11_depth_stencil_view_cb: sapp.d3d11_get_depth_stencil_view
 	}
 	gfx.setup(&desc)
 	sgl_desc := sgl_desc_t{}
@@ -138,20 +138,20 @@ pub fn (gg &GG) run() {
 pub fn (ctx &GG) draw_rect(x, y, w, h f32, c gx.Color) {
 	sgl.c4b(c.r, c.g, c.b, 128)
 	sgl.begin_quads()
-	C.sgl_v2f(x, y)
-	C.sgl_v2f(x + w, y)
-	C.sgl_v2f(x + w, y + h)
-	C.sgl_v2f(x, y + h)
+	sgl.v2f(x, y)
+	sgl.v2f(x + w, y)
+	sgl.v2f(x + w, y + h)
+	sgl.v2f(x, y + h)
 	sgl.end()
 }
 
 pub fn draw_rect(x, y, w, h f32, c gx.Color) {
 	sgl.c4b(c.r, c.g, c.b, 128)
 	sgl.begin_quads()
-	C.sgl_v2f(x, y)
-	C.sgl_v2f(x + w, y)
-	C.sgl_v2f(x + w, y + h)
-	C.sgl_v2f(x, y + h)
+	sgl.v2f(x, y)
+	sgl.v2f(x + w, y)
+	sgl.v2f(x + w, y + h)
+	sgl.v2f(x, y + h)
 	sgl.end()
 }
 

